### ISSUE TITLE
Programatically Update Segment without Touch?

### ISSUE THREAD

**Issue by KjellConnelly** [kirankalyan5/react-native-segmented-control-tab#6]
I have a situation where I need to toggle the segment which is selected based on some logic. But I can't figure out how to change the ControlTab's actual UI to reflect the state, as selectedIndex seems to only affect the initially selected index, but doesn't update dynamically. 

Any way around this?

**Comment by SuhairZain** [comment]
Hi @KjellConnelly 
The way this component is currently designed, the selectedIndex is only for the initial selection, with the further changes manages internally using state.

It's not a workaround, but you could rewrite this component to be a controlled component, with no internal state. You could do it yourself, it's fairly easy. Otherwise, if you can afford to wait, I'll be submitting a PR for the same by tomorrow.

**Comment by KjellConnelly** [comment]
Cool. Hows the progress going? And thanks :)

**Comment by SuhairZain** [comment]
@KjellConnelly Damn, dude. I was caught up in another work and totally forgot about it. I shall get to it right now. Sorry for the delay.

**Comment by kirankalyan5** [comment]
@KjellConnelly You will be getting to see some updates today.

Thanks @SuhairZain for the PR 👍 . Cheers!

**Comment by kirankalyan5** [comment]
@SuhairZain Merged and updated the example as well, see if you can make the example more crisp.

**Comment by SuhairZain** [comment]
Hi @kirankalyan5 
This issue was for a stateless tab selection, not multiple selection. I'm yet to make a PR for it. Kindly re-open the issue. 😄

**Comment by SuhairZain** [comment]
Hi @KjellConnelly 
I've made it to a controlled component but I'm yet to update the documentation and example. For the time being, until I do that, you can use [my fork](https://github.com/SuhairZain/react-native-segmented-control-tab/).

@kirankalyan5 I realize that this is a really bad way to do things and I'm really sorry about it. I'll write the documentation and examples as soon as I can and submit a PR, but this is just for the user's immediate use.

**Comment by kirankalyan5** [comment]
@SuhairZain No problem buddy, re opening the issue.

**Comment by jvt** [comment]
Has there been any update on this?

**Comment by kirankalyan5** [comment]
Hi @jvt, I have been busy for a while. I have few changes, which I will be pushing them soon!
Between you can refer @SuhairZain [fork](https://github.com/SuhairZain/react-native-segmented-control-tab/) for the above update.

**Comment by mojocakes** [comment]
Not ideal, but you can do it like this:

Save a reference to the component
```javascript
<SegmentedControlTab ref="segmentedControl" ... />
```

Invoke the component's internal "onTabPress" function and pass it a tab index
```javascript
this.refs.segmentedControl.onTabPress(0)
```

**Comment by SuhairZain** [comment]
I've submitted a PR with the changes. Once the author merges them, you can
use it directly without any hacks.

On Mar 31, 2017 4:02 PM, "James Cook" <notifications@github.com> wrote:

Not ideal, but you can do it like this:

Save a reference to the component

<SegmentedControlTab ref="segmentedControl" ... />

Invoke the component's internal "onTabPress" function and pass it a tab
index

this.refs.segmentedControl.onTabPress(0)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kirankalyan5/react-native-segmented-control-tab/issues/6#issuecomment-290678251>,
or mute the thread

**Comment by kirankalyan5** [comment]
@SuhairZain I have merged the changes and published 3.0.1 👍 

@here - have a look on the latest code.
